orchagent/fgnhgorch.cpp

```diff
@@ -207,23 +207,30 @@ void FgNhgOrch::setStateDbRouteEntry(const IpPrefix &ipPrefix, uint32_t index, N
     // check if profile already exists - if yes - skip creation
     m_stateWarmRestartRouteTable.get(key, fvs);
 
-    //bucket rewrite
-    if (fvs.size() > index)
+    bool isUpdated = false;
```

Comment:
Instead of this logic, why don't we just do an hset(as opposed to the get +set) on the table with key as prefix, and fv pair as <index, nh key>.